enterprise/server/backends/redis_client/redis_client.go

```diff
@@ -0,0 +1,69 @@
+package redis_client
+
+import (
+	"context"
+	"flag"
+
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/util/redisutil"
+	"github.com/buildbuddy-io/buildbuddy/server/config"
+	"github.com/buildbuddy-io/buildbuddy/server/environment"
+	"github.com/buildbuddy-io/buildbuddy/server/util/flagutil"
+	"github.com/buildbuddy-io/buildbuddy/server/util/status"
+)
+
+var (
+	defaultRedisTarget          = flag.String("app.default_redis_target", "", "A Redis target for storing remote shared state. To ease migration, the redis target from the remote execution config will be used if this value is not specified.")
+	defaultRedisShards          = flagutil.StringSlice("app.default_sharded_redis.shards", []string{}, "Ordered list of Redis shard addresses.")
+	defaultShardedRedisUsername = flag.String("app.default_sharded_redis.username", "", "Redis username")
+	defaultShardedRedisPassword = flag.String("app.default_sharded_redis.password", "", "Redis password")
+)
+
+func DefaultRedisClientConfigNoFallback() *config.RedisClientConfig {
+	if len(*defaultRedisShards) > 0 {
+		return &config.RedisClientConfig{
+			ShardedConfig: &config.ShardedRedisConfig{
+				Shards:   *defaultRedisShards,
+				Username: *defaultShardedRedisUsername,
+				Password: *defaultShardedRedisPassword,
+			},
+		}
+	}
+
+	if *defaultRedisTarget != "" {
+		return &config.RedisClientConfig{SimpleTarget: *defaultRedisTarget}
+	}
+	return nil
+}
+
+func DefaultRedisClientConfig(env environment.Env) *config.RedisClientConfig {
+	if drcc := DefaultRedisClientConfigNoFallback(); drcc != nil {
```

Comment:
uber nit: DefaultRedisClientConfig isn't a common acronym (yet?) so IMO something like `cfg`, `conf`, or `config` would be a more easily grokkable abbreviation (same below)